components/renku_data_services/base_models/core.py

```diff
@@ -51,12 +51,24 @@ class GitlabAPIUser(APIUser):
     name: str
     gitlab_url: str
 
+    @property
+    def gitlab_graphql_url(self):
+        """Gets the graphql url for gitlab."""
+        gitlab_url = self.gitlab_url
+
+        if not gitlab_url.startswith("http") and "://" not in gitlab_url:
+            gitlab_url = f"https://{gitlab_url}"
+
+        gitlab_url = gitlab_url.rstrip("/")
```

Comment:
I think it is better to check whether gitlab_url is a valid/proper url in a `__post_init__` method of thi dataclass and crash if you dont like what you see. You can also run the `.rstrip` to remove trailing slashes there. Then you can just freely use the value in other methods/places.